### ISSUE TITLE
Support sections with no keys

### ISSUE THREAD

**Issue by dimsuz** [QEDK/configparser-rs#37]
Currently when I read this file:

```ini
[section0]

[section1]
key = value
```
the resulting map (and `conf.sections()` too) will only contain `section1`. 
`section0` will be silently dropped.

I didn't find any way to configure this crate so that sections with no keys are not skipped.

There are some use-cases where one wants to know if section is present (even if empty).

**Comment by QEDK** [comment]
This is an interesting edge-case, let me look into the parser implementation.

**Comment by QEDK** [comment]
The issue would arise from our handling of section headers, https://github.com/QEDK/configparser-rs/blob/1098ef7e6c08abaa3839a67e9830acfc032022f7/src/ini.rs#L486 we expect a non-empty map for each section, a section header followed by another section header means nothing is added to the map. This would be breaking change, what's the breaking use-case atm? @dimsuz The only relative way of fixing this would be using `HashMap<String, Option<HashMap<String, Option<String>>>>`.

**Comment by dimsuz** [comment]
My use-case is a bit of a weird one actually.

I'm using this library to parse an "ini-like" file format, which uses double-brackets to signify a higher level section, for example this:

```
[[Section1]]
[subsectionA]
key = value

[[Section2]]
[subsectionB]
key = value
```
should result in this hierarchy after parsing:

```mermaid
graph TD;
    Section1-->subsectionA-->keys1{{key=value}};
    Section2-->subsectionB-->keys2{{key=value}};
```
This is not strictly INI, but `configparser` is still able to parse it and my idea was to look for sections with name `[Section1]` (one level of `[]` gets stripped by the lib) and that would do it - I would be able to build parent-child relationships from there, sections being empty is OK, I would use indexed map and rely on order.

Currently this not working is not a big issue for me, so if it's impacting a library design or backwards compatibility I guess it's fine to close this issue and I'll have to stop being lazy and write some proper parser instead :)

**Comment by belak** [comment]
I've got an implementation in https://github.com/QEDK/configparser-rs/pull/50. I was able to do it without changing the API by just creating an empty HashMap when we encounter the section header, which seems clear enough.

Our use case is that having sections means that something is enabled, and allowing for extra configuration as options in that section. An empty section would mean it was enabled but there was no additional configuration.
